Teapot Implementations & Fixes

NotThorny opened the pull request:
## Description

Fixes and additional implementations for the teapot system.

- [x] Unlocking realms from their respective items
- [x] Switching between realms
- [x] Exiting the teapot to where you originally entered
- [x] Claim home level rewards
- [x] Fixes Grass main house bricking accounts
- [x] Fixes Grass realm due to house fix (Grass realm only: main house needs to be placed manually)
- [x] Adds gaining trust (realm exp) from crafting and with `/give`
- [x] Make trust values show in djinn menu

Example of some of the functions:
https://streamable.com/nrp03d

## Issues fixed by this PR

<!--- Put the links of issues that may be fixed by this PR here (if any). -->

Fixes #2051 

## Type of changes

<!--- Put an `x` in all the boxes that apply your changes. -->

- [x] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

NotThorny committed:
Add realm switching
Fix realm unlock items

NotThorny committed:
Implement exiting teapot

NotThorny committed:
Implement home level rewards

NotThorny committed:
Fix small issues

NotThorny committed:
Fix call issue

NotThorny committed:
Add 'seen unlock' packets

NotThorny committed:
Merge branch 'development' of https://github.com/NotThorny/Grasscutter into development

NotThorny committed:
Fix Sumeru Main House bricking accounts
This is only for the house, not the full realm

NotThorny committed:
Fix Grass main house bricking accounts

NotThorny committed:
Fixed realm 5

Main house needs to be placed manually for Sumeru realm (module 5) as the resources for defaults in Sumeru realm are missing. Sumeru main house cannot be exited from the door, issue due to the same missing res.

NotThorny committed:
Merge branch 'development' of https://github.com/NotThorny/Grasscutter into development

NotThorny committed:
Remove references

KingRainbow44 committed:
Formatting change (web editor)

KingRainbow44 commented on src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseUnlockHomeModule.java:
```diff
@@ -14,6 +14,13 @@ public ItemUseUnlockHomeModule(String[] useParam) {
+
+    @Override
+    public boolean postUseItem(UseItemParams params){
+
```
remove whitespace

NotThorny replied:
Whitespace and formatting should be fixed now.

KingRainbow44 commented on src/main/java/emu/grasscutter/server/packet/recv/HandlerBackMyWorldReq.java:
```diff
@@ -0,0 +1,34 @@
+	
+	@Override
+	public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+
```
whitespace

KingRainbow44 commented on src/main/java/emu/grasscutter/server/packet/recv/HandlerHomeChangeModuleReq.java:
```diff
@@ -0,0 +1,38 @@
+import emu.grasscutter.server.packet.send.PacketHomeComfortInfoNotify;
+import emu.grasscutter.server.packet.send.PacketPlayerHomeCompInfoNotify;
+import emu.grasscutter.utils.Position;
+
```
spacing

KingRainbow44 commented on src/main/java/emu/grasscutter/server/packet/send/PacketPlayerEnterSceneNotify.java:
```diff
@@ -43,6 +43,14 @@ public PacketPlayerEnterSceneNotify(Player player, EnterType type, EnterReason r
         super(PacketOpcodes.PlayerEnterSceneNotify);
 
+        // Set previous position
+        if(!(newScene == 3)){ // Hardcoded for now else weird positions will occur
+            // Don't update position within same scene or teapot
+        }else{
```
Fix spacing.
`if (`
`} else {`

KingRainbow44 reviewed (dismissed):
i feel rlly bad if whitespace bot goes in automatically and fixes whitespace lol

NotThorny committed:
Whitespace & Formatting

NotThorny committed:
Whitespace

NotThorny committed:
Make 'seen' status persistent

NotThorny committed:
Fix misnamed field

NotThorny committed:
Revert "Fix misnamed field"

This reverts commit 21ef404e14b3111db47c25a15ba86bbd2eaab6db.

NotThorny committed:
Implement gaining trust (realm exp)

Add gaining trust by crafting
Add gaining trust via `/give`
Show trust in djinn menu

NotThorny committed:
Interior check for prevScene

NotThorny committed:
Correct positions & rotations

Return to front of main house when exiting from inside

GanyusLeftHorn committed:
Update HandlerBackMyWorldReq.java
